### PR TITLE
Post Settings: Post Format

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR updates the Post Format picker and makes several improvements to how it is setup. This is also the first PR that directly updates the `PostModel` after a selection.

We decided to keep the `PostModel` and the DB representation of it up to date in Post Settings. So, in the next PR, I'll update the rest of the settings to dispatch the changes as soon as it happens. Please treat this PR as a work-in-progress still, as there will be couple of changes in the next PR to it. However, aside from how/when the information is saved, it should be close to it's final form.

To test:
* Go into Post Settings and select the Post Format and play with different scenarios. Make sure the correct format is initially selected and when another option is selected, it's updated in the UI correctly.

/cc @tonyr59h

### CODE REVIEW

**Comment by tonyr59h** [comment]
When I created a new post and went to the settings screen there was no default format selected. The dialog had the correct format selected though, and when I click `OK` the view is correctly updated.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -977,6 +977,7 @@
     <string name="editor_post_settings_featured_image">Featured Image</string>
     <string name="editor_post_settings_set_featured_image">Set Featured Image</string>
     <string name="post_settings_status">Status</string>
+    <string name="post_settings_format">Post Format</string>
```
Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: We can get rid of `post_format` since this is being used instead.

**Comment by tonyr59h** [comment]
Aside from the bug mentioned above and the unused string resource this PR is in good shape. Great work @oguzkocer!

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (oguzkocer) [inline WordPress/src/main/res/values/strings.xml]: Fixed in a8ef3a61e2b708223288500d4638529fba718852. I blame AS for not graying out the unused resource 🤒

**Comment by oguzkocer** [comment]
@tonyr59h I have resolved the merge conflicts and fixed the string resource issue. It should be ready for another pass!

> When I created a new post and went to the settings screen there was no default format selected. The dialog had the correct format selected though, and when I click OK the view is correctly updated.

Opened a new issue for this #6044 as it's not really a UI issue and out of the scope of this PR. If we can't properly fix the #6044 we can consider adding a "Not Set" label, however we should try to do it the proper way first.

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**